runners/spark/src/test/java/org/apache/beam/runners/spark/translation/streaming/StreamingSourceMetricsTest.java

```diff
@@ -47,11 +52,18 @@
   @Test
   @Category(StreamingTest.class)
   public void testUnboundedSourceMetrics() {
-    final long numElements = 1000;
+    final long minElements = 1000;
 
-    pipeline.apply(
-        // Use maxReadTime to force unbounded mode.
-        GenerateSequence.from(0).to(numElements).withMaxReadTime(Duration.standardDays(1)));
+    // Use a GenerateSequence for the UnboundedSequence, but push the watermark to infinity at
+    // minElements to let the test pipeline cleanly shut it down.  Shutdown will occur shortly
+    // afterwards, but at least minElements will be reported in the metrics.
+    PCollection<Long> pc =
+        pipeline.apply(
+            GenerateSequence.from(1)
+                .withRate(minElements / 10, Duration.millis(500L))
+                .withTimestampFn(
+                    t -> t < minElements ? Instant.now() : BoundedWindow.TIMESTAMP_MAX_VALUE));
+    assertThat(pc.isBounded(), is(PCollection.IsBounded.UNBOUNDED));
```

Comment:
Note: this assert would have failed with the old GenerateSequence configuration.